src/YourEasyRent/Services/State/UserStateManager.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace YourEasyRent.Services.State
+{
+    public class UserStateManager : IUserStateManager
+    {
+        public bool IsCategoryChosen { get; private set; }
+        public bool IsBrandChosen { get; private set; }
+        public bool IsBackOnOneStep { get; private set; }
+        public bool IsReturnToMainMenu { get; private set; }
+
+        private List<string> menuStatuses = new List<string>();
+        public void Start()
+        {
+            menuStatuses.Add("Started");
+        }
+        public void SetBrand(MenuStatus status)
+        {
+            IsBrandChosen = true;
+            menuStatuses.Add("BrandChosen");
+        }
+        public void SetCategory(MenuStatus status)
+        {
+            IsCategoryChosen = true;
+            menuStatuses.Add("CategoryChosen");
+        }
+        public void GoBackToOneStep(string status)
+        {
+            IsBackOnOneStep = true;
+            menuStatuses.Last();
+            menuStatuses.Remove(status);
```

Comment:
Just remove the last element in `menuStatuses`
`menuStatuses.RemoveAt(numbers.Count - 1);`